packages/core/src/browser/preferences/preference-provider.ts

```diff
@@ -5,11 +5,28 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
+import { injectable } from 'inversify';
 import { Disposable, Event } from '../../common';
 
 export const PreferenceProvider = Symbol('PreferenceProvider');
 
 export interface PreferenceProvider extends Disposable {
```

Comment:
Would you consider to use `PreferenceProvider` as a stub implementation by default? It allows to get rid of interface + symbol and use a class directly as a class identifier.

```ts
import { injectable } from 'inversify';
import { DisposableCollection, Disposable, Event, Emitter } from '../../common';

@injectable()
export class PreferenceProvider implements Disposable {

    protected readonly onDidPreferencesChangedEmitter = new Emitter<void>();
    readonly onDidPreferencesChanged: Event<void> = this.onDidPreferencesChangedEmitter.event;

    protected readonly toDispose = new DisposableCollection();

    constructor() {
        this.toDispose.push(this.onDidPreferencesChangedEmitter);
    }

    dispose(): void {
        this.toDispose.dispose();
    }

    protected fireOnDidPreferencesChanged(): void {
        this.onDidPreferencesChangedEmitter.fire(undefined);
    }

    getPreferences(): { [p: string]: any } {
        return [];
    }

    setPreference(key: string, value: string): Promise<void> {
        return Promise.resolve();
    }

}
```

Reply:
done